packages/http/src/validator/validators/body.ts

```diff
@@ -139,3 +53,95 @@ function decodeUriEntities(target: Dictionary<string, string>) {
     return result;
   }, {});
 }
+
+function findContentByMediaTypeOrFirst(specs: IMediaTypeContent[], mediaType: string) {
+  return pipe(
+    specs,
+    Array.findFirst(spec => spec.mediaType === mediaType),
+    Option.alt(() => Array.head(specs)),
+    Option.map(content => ({ mediaType, content })),
+  );
+}
+
+function validateBodyIfNotFormEncoded(mediaType: string, schema: JSONSchema, target: unknown) {
+  return pipe(
+    mediaType,
+    Option.fromPredicate(mt => !!!typeIs.is(mt, ['application/x-www-form-urlencoded'])),
+    Option.chain(() => Option.some(validateBody(schema, target))),
+  );
+}
+
+function deserializeAndValidate(content: IMediaTypeContent, schema: JSONSchema, target: string) {
+  const encodings = get(content, 'encodings', []);
+  const encodedUriParams = splitUriParams(target);
+
+  return pipe(
+    validateAgainstReservedCharacters(encodedUriParams, encodings),
+    Either.map(decodeUriEntities),
+    Either.map(decodedUriEntities => deserializeFormBody(schema, encodings, decodedUriEntities)),
+    Either.fold(e => Option.some(e), deserialised => Option.some(validateBody(schema, deserialised))),
+  );
+}
+
+export class HttpBodyValidator implements IHttpValidator<any, IMediaTypeContent> {
+  constructor(private prefix: string) {}
+
+  public validate(target: any, specs: IMediaTypeContent[], mediaType?: string): IPrismDiagnostic[] {
+    const mediaTypeWithContentAndSchema = pipe(
+      Option.fromNullable(mediaType),
+      Option.chain(mt => findContentByMediaTypeOrFirst(specs, mt)),
+      Option.alt(() => Option.some({ content: specs[0] || {}, mediaType: 'random' })),
```

Comment:
I think we should keep `mediaType` optional here. It is optional on the input level and it's better to keep it like that.